genai-perf/tests/integration_tests/test_multiturn.py

```
@@ -0,0 +1,92 @@
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import json
+from pathlib import Path
+from unittest.mock import mock_open, patch
+
+import pytest
+from genai_perf.inputs.input_constants import PromptSource
+from genai_perf.inputs.inputs import Inputs
+from genai_perf.inputs.inputs_config import InputsConfig
+from genai_perf.inputs.retrievers.generic_dataset import GenericDataset
+from genai_perf.inputs.retrievers.payload_input_retriever import PayloadInputRetriever
+from genai_perf.tokenizer import get_empty_tokenizer
+
+
+class TestIntegrationMultiTurn:
+    @pytest.fixture
+    def mock_config(self):
+        class MockConfig(InputsConfig):
+            def __init__(self):
+                self.tokenizer = get_empty_tokenizer()
+                self.model_name = ["test_model"]
+                self.payload_input_filename = Path("test_input.jsonl")
+                self.output_dir = Path("./test_output")
+                self.random_seed = 42
+                self.prompt_tokens_mean = 10
+                self.input_type = PromptSource.PAYLOAD
+                self.extra_inputs = {}
+
+        return MockConfig()
+
+    @pytest.fixture
+    def retriever(self, mock_config):
+        return PayloadInputRetriever(mock_config)
+
+    def test_multi_turn_input__generation(self, monkeypatch, mock_config):
+        input_data = (
+            '{"text": "Turn 1", "session_id": "abc", "delay": 2}\n'
+            '{"text": "Turn 2", "session_id": "abc", "delay": 3}\n'
+            '{"text": "Turn 3", "session_id": "abc", "timestamp": 5000}\n'
```

Is this functionally correct? 
The way multi-turn is implemented today, we can specify a mix of delay and timestamp and PA ignores a request if it doesn't contain "delay" in multi-turn mode.
However, I think GAP should throw an error in this case instead of PA silently ignoring the request. 
The test is fine in itself but I think this validation should be added in the feature.

---

I see. I updated the unit test. Thanks for pointing this out!

I think PA today returns the error and GAP forwards it to the user, but we could always run it and validate it.